rpc/rpcserver/server.go

```diff
@@ -682,6 +682,69 @@ func (s *walletServer) StakeInfo(ctx context.Context, req *pb.StakeInfoRequest)
 	}, nil
 }
 
+// makeChangeSourceFromAddress creates a ChangeSource which is used to
+// receive all correlated previous input value.
+func makeChangeSourceFromAddress(address string) txauthor.ChangeSource {
+	return func() ([]byte, uint16, error) {
+		destinationAddress, err := dcrutil.DecodeAddress(address)
+		if err != nil {
+			return nil, 0, err
+		}
+		script, err := txscript.PayToAddrScript(destinationAddress)
+		if err != nil {
+			return nil, 0, err
+		}
+
+		return script, txscript.DefaultScriptVersion, err
+	}
+}
+
+func (s *walletServer) SweepAccount(ctx context.Context, req *pb.SweepAccountRequest) (*pb.SweepAccountResponse, error) {
+	// use provided fee per Kb if specified.
+	feePerKb := s.wallet.RelayFee()
+	if req.FeePerKb > 0 {
```

Comment:
this should check if the relay fee is non-zero.  There should also be an additional check which returns an error if the value is negative.